code/API_definitions/qod-api.yaml

```
@@ -290,6 +290,66 @@ paths:
         - threeLegged:
             - "qod-sessions-delete"
 
+  /sessions/{sessionId}/extend:
+    post:
+      tags:
+        - Change session duration
+      summary: "Extend the duration of an active session"
+      description: |
+        Extend the duration of an active QoS session.
+        If this operation is executed successfully, the new duration of the target session will be the original duration plus extended duration.
+      operationId: extendQosSessionDuration
+      parameters:
+        - name: sessionId
+          in: path
+          description: Session ID that was obtained from the createSession operation
+          required: true
+          schema:
+            $ref: "#/components/schemas/SessionId"
+      requestBody:
+        description: Parameters to extend the duration of an active session
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ExtendSessionDuration"
+        required: true
+      responses:
+        "200":
+          description: Contains information about active session
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SessionInfo"
+        "400":
+          description: Invalid input for extendQosSessionDuration operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              examples:
+                Generic400:
+                  summary: Some parameter combinations or parameter values provided are not schema compliant
+                  value:
+                    status: 400
+                    code: INVALID_ARGUMENT
+                    message: "Schema validation failed at  ..."
+                InactiveSession:
+                  summary: The target session is inactive
+                  value:
+                    status: 400
+                    code: INVALID_ARGUMENT
+                    message: "The target session is inactive"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/SessionNotFound404"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+
```

```suggestion

      security:
        - oAuth2ClientCredentials: []
        - threeLegged:
            - "qod-sessions-write"
```

The scope for write is shared with the operation to create session. We could differentiate between -create and -update but -write covers both.

---

It is added.